### ISSUE TITLE
Ability to receive/wait for nacked messages, pause a topic

### ISSUE THREAD

**Issue by sneerin** [apache/pulsar-client-go#351]
Currently, if you send 5 messages, and back #5 , it will arrive the last one, breaking the order.
i want to have a use-case, when once a message nacked, the order is not broken, the topic is on pause, until the message is returned or completely failed to DLQ.

**Comment by jiazhai** [comment]
if you are using exclusive/failover sub mode, the order should be right. Are you using shared mode?

**Comment by sneerin** [comment]
No, I'm using exclusive mode, still, I have a message that I nacked only after the rest of them arrived, I'm doing nack 2 times.
